### PR TITLE
Add error id to exceptions

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add error id (like `BAD_ARGUMENTS`) to exception messages. This closes #25862.

### CODE REVIEW

File: programs/client/Client.cpp [hunk 1]
```diff
@@ -2597,7 +2565,7 @@ class Client : public Poco::Util::Application
             catch (const Exception & e)
             {
                 std::string text = e.displayText();
```
Comment (vdimir) [inline programs/client/Client.cpp:2567]:
```suggestion
```

File: src/Server/GRPCServer.cpp [hunk 2]
```diff
@@ -1150,7 +1150,7 @@ namespace
     {
         io.onException();
 
-        LOG_ERROR(log, "Code: {}, e.displayText() = {}, Stack trace:\n\n{}", exception.code(), exception.displayText(), exception.getStackTraceString());
+        LOG_ERROR(log, "{}", getExceptionMessage(exception, true));
```
Comment (vdimir) [inline src/Server/GRPCServer.cpp:1153]: Why not:
```suggestion
        LOG_ERROR(log, getExceptionMessage(exception, true));
```
?

File: src/Server/TCPHandler.cpp [hunk 3]
```diff
@@ -149,7 +149,7 @@ void TCPHandler::runImpl()
         if (!DatabaseCatalog::instance().isDatabaseExist(default_database))
         {
             Exception e("Database " + backQuote(default_database) + " doesn't exist", ErrorCodes::UNKNOWN_DATABASE);
-            LOG_ERROR(log, "Code: {}, e.displayText() = {}, Stack trace:\n\n{}", e.code(), e.displayText(), e.getStackTraceString());
+            LOG_ERROR(log, "{}", getExceptionMessage(e, true));
```
Comment (vdimir) [inline src/Server/TCPHandler.cpp:152]:
```suggestion
            LOG_ERROR(log, getExceptionMessage(e, true));
```

File: src/Server/TCPHandler.cpp [hunk 4]
```diff
@@ -422,7 +422,7 @@ void TCPHandler::runImpl()
                 }
 
                 const auto & e = *exception;
-                LOG_ERROR(log, "Code: {}, e.displayText() = {}, Stack trace:\n\n{}", e.code(), e.displayText(), e.getStackTraceString());
+                LOG_ERROR(log, "{}", getExceptionMessage(e, true));
```
Comment (vdimir) [inline src/Server/TCPHandler.cpp:425]:
```suggestion
                LOG_ERROR(log, getExceptionMessage(e, true));
```

**Review by vdimir — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Only `test_replicated_fetches_timeouts/test.py::test_no_stall` remains.

**Comment by vdimir** [comment]
Also there's failures in TestFlows, I've looked at it, but seems that at least some of them are not related to exception messages, I didn't go deep into it.

**Comment by vdimir** [comment]
> Integration tests flaky check (asan)

Seems not related 
`test_storage_kafka`

```
>                   raise error_type(
                        "Request '{}' failed with response '{}'."
                        .format(request, response))
E                   kafka.errors.UnknownError: [Error -1] UnknownError: Request 'CreateTopicsRequest_v3(create_topic_requests=[(topic='consumer_hang2', num_partitions=1, replication_factor=1, replica_assignment=[], configs=[])], timeout=30000, validate_only=False)' failed with response 'CreateTopicsResponse_v3(throttle_time_ms=0, topic_errors=[(topic='consumer_hang2', error_code=-1, error_message='KeeperErrorCode = Session expired for /brokers/topics/consumer_hang2')])'.
```

> Integration tests (release)

Also looks like not related

`test_MemoryTracking/test.py::test_tcp_multiple_sessions`

```
>       assert changed < changes_allowed
E       assert 16 < 10

test_MemoryTracking/test.py:62: AssertionError
```

`test_jdbc_bridge`

```
        if (self.process.returncode != 0 or stderr) and not self.ignore_error:
>           raise QueryRuntimeException(
                'Client failed! Return code: {}, stderr: {}'.format(self.process.returncode, stderr), self.process.returncode, stderr)
E           helpers.client.QueryRuntimeException: Client failed! Return code: 86, stderr: Received exception from server (version 21.9.1):
E           Code: 86. DB::Exception: Received from 172.16.17.3:9000. DB::Exception: Received error from remote server /columns_info?connection_string=self%3Fmutation&table=SET%20mutations_sync%20%3D%201%3B%20ALTER%20TABLE%20test.test_delete%20DELETE%20WHERE%20Num%20%3C%201000%20-%201%3B&external_table_functions_use_nulls=true. HTTP status code: 500 Internal Server Error, body: NamedDataSource [self] does not exist!. Stack trace:
E
```

> Functional stateless tests flaky check (address) — Timeout

No errors in log, but it failed in previous commit to, can be related?

**Pull request merged**